cast/src/lib.rs

```diff
@@ -1656,6 +1662,75 @@ impl SimpleCast {
     pub fn disassemble(bytecode: &str) -> Result<String> {
         format_operations(disassemble_str(bytecode)?)
     }
+
+    /// Gets the selector for a given function signature
+    /// Optimizes if the `optimize` parameter is set to a number of leading zeroes
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::get_selector(String::from("foo(address,uint256)"), None)?.0, String::from("0xbd0d639f"));
+    ///     assert_eq!(Cast::get_selector(String::from("foo(address,uint256)"), Some(2))?.0, String::from("0x0000bf3e"));
+    ///     assert_eq!(Cast::get_selector(String::from("foo(address,uint256)"), Some(2))?.1, Some(String::from("foo71661(address,uint256)")));
+    ///
+    ///     Ok(())
+    /// }
+    /// ```    
+    pub fn get_selector(
+        signature: String,
+        optimize: Option<usize>,
+    ) -> Result<(String, Option<String>, Option<Duration>)> {
+        if optimize.is_none() || optimize.unwrap() > 4 {
+            let selector = HumanReadableParser::parse_function(&signature)?.short_signature();
+            return Ok((format!("0x{}", hex::encode(selector)), None, None))
+        }
+        let mut name: &str = "";
+        let mut params: &str = "";
+        if let Some(index) = signature.find('(') {
+            name = &signature[..index];
+            params = &signature[index..];
+        }
```

Comment:
This should return `Err` if `None` instead of carrying empty/invalid values.

[`str::split_once()`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.split_once) is more idiomatic for this, though it will exclude the separator so if you add this you will need to add `(` to the signature format string.

```suggestion
        let Some((name, params)) = signature.split_once('(') else {
            eyre::bail!("Invalid signature");
        }
```